catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -524,12 +538,68 @@ public void init() {
         APPLICATION_JSON,
         (req, res) -> {
           String id = req.params(":id");
+
+          final Metacard workspace = getWorkspaceMetacard(id);
+          Attribute queries =
+              workspace.getAttribute(WorkspaceConstants.WORKSPACE_ASSOCIATED_QUERIES);
+
+          if (queries != null) {
+            String[] queryIds =
+                queries.getValues().stream().map(String.class::cast).toArray(String[]::new);
+
+            catalogFramework.delete(new DeleteRequestImpl(queryIds));
+          }
+
           catalogFramework.delete(new DeleteRequestImpl(id));
           subscriptions.removeSubscriptions(id);
           return ImmutableMap.of("message", "Successfully deleted.");
         },
         util::getJson);
 
+    post(
```

Comment:
Add an endpoint for retrieving all query metacards in a workspace and add paging